primitives/version/src/lib.rs

```diff
@@ -207,8 +300,33 @@ pub struct RuntimeVersion {
 	///
 	/// It need *not* change when a new module is added or when a dispatchable is added.
 	pub transaction_version: u32,
+
+	/// Version of the state implementation used by this runtime.
+	/// Use of an incorrect version is consensus breaking.
+	pub state_version: u32,
```

Comment:
Why not use the enum here directly?

Reply:
I started doing it, but this also involve some macro changes (for the decl_version macro).
Currently we are using:
```
			parse_once(&mut self.state_version, field_value, Self::parse_num_literal)?;
```
but with state version the will be need to parse state version, is it worth it?

Reply:
Ahh okay. Yeah then drop it.

Do we actually need to make this `u32`? Or can we just make this `u8`?

Reply:
yes u8 is also fine (could change format if it becomes an issue someday)